src/models/MessageRepository.ts

```diff
@@ -0,0 +1,239 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+/* eslint-disable class-methods-use-this */
+
+import {
+  DataTypes, Model, InitOptions, Op,
+} from 'sequelize';
+
+export class MessageModel extends Model {}
+
+/**
+ * To be used for calling any methods which would change states of record(s) in Database.
+ */
+export interface MessageAttributes {
+  messageHash: string;
+  type: string;
+  gatewayAddress: string;
+  sourceStatus: string;
+  targetStatus: string;
+  gasPrice: number;
+  gasLimit: number;
+  nonce: number;
+  sender: string;
+  direction: string;
+  sourceDeclarationBlockHeight: number;
+  secret?: string;
+  hashLock?: string;
+}
+
+/**
+ * Repository would always return database rows after typecasting to this
+ */
+export interface Message extends MessageAttributes{
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+export class MessageConstant {
+  /* Stake and mint type. */
+  public static stakeAndMintType: string = 'stakeAndMint';
+
+  /* Redeem and unstake type. */
+  public static redeemAndUnstakeType: string = 'redeemAndUnstake';
+
+  /* Undeclared status. */
+  public static unDeclaredStatus: string = 'Undeclared';
+
+  /* Declared status. */
+  public static declaredStatus: string = 'Declared';
+
+  /* Progressed status. */
+  public static progressedStatus: string = 'Progressed';
+
+  /* RevocationDeclared status. */
+  public static revocationDeclaredStatus: string = 'RevocationDeclared';
+
+  /* Revoked status. */
+  public static revokedStatus: string = 'Revoked';
+
+  /* Statuses which are used for sourceStatus & targetStatus. */
+  public static statusesArray: string[] = [
+    MessageConstant.declaredStatus,
+    MessageConstant.unDeclaredStatus,
+    MessageConstant.progressedStatus,
+    MessageConstant.revocationDeclaredStatus,
+    MessageConstant.revokedStatus,
+  ];
+
+  /* Origin to auxiliary direction. */
+  public static originToAuxiliaryDirection: string = 'o2a';
+
+  /* Auxiliary to origin direction. */
+  public static auxiliaryToOriginDirection: string = 'a2o';
+}
```

Comment:
These all can be ENUMs
```
enum MessageType {
    Stake = 'stake',
    Redeem = 'redeem',
}
```    
```
enum MessageStatus {
    Undeclared = 'undeclared',
    Declared = 'declared',
    Progressed = 'progressed',
    RevocationDeclated = 'revocation_declared',
    Revoked = 'revoked',
}
```
```
enum MessageDirection {
    OriginToAuxuliary = 'o2a',
    AuxiliaryToOrigin = 'a2o',
}

```

Reply:
This is much better 👍. Have changed usages accordingly.